src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -69,16 +72,20 @@ public function process(ContainerBuilder $container)
 
         $this->registerReceivers($container);
         $this->registerSenders($container);
-        $this->registerHandlers($container);
+        $this->registerHandlers($container, $buses);
     }
 
-    private function registerHandlers(ContainerBuilder $container)
+    private function registerHandlers(ContainerBuilder $container, array $buses)
     {
         $definitions = array();
-        $handlersByMessage = array();
+        $handlersByBusAndMessage = array();
 
         foreach ($container->findTaggedServiceIds($this->handlerTag, true) as $serviceId => $tags) {
             foreach ($tags as $tag) {
+                if (isset($tag['bus']) && !\in_array($tag['bus'], $buses, true)) {
```

Comment:
for being service ids', in theory it could resolve aliases. Allowing `bus: message_bus` instead of `bus: messenger.bus.default`.

Personally i'd find that useful for handlers registered by 3rd party bundles, allowing them to use a fixed alias id during registration and let the user provide the alias.

Reply:
No strong opinion on this. Let's see what others think :)
Perhaps not in this PR anyway in order to not add more complexity.

Reply:
In my case im still hesitating between auto registration and documenting the user config. Things like this and removing the middleware tag makes me lean to the latter :thinking:

Reply:
You can already use `bus: command_bus` for example if you define your bus that way:
```yaml
framework:
    messenger:
        buses:
            command_bus: ~
```